src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -353,7 +353,7 @@ private static void CleanupSourceMappingStart(FormattingContext context, Range s
                 // so @{    throw null; } will be formatted to @{ throw null; }
                 // Ideally we'd put that across three lines, which is what normal formatting does, but since we
                 // can't control the cursor, that doesn't end well.
-                if (isOnType)
+                if (isOnType && sourceMappingRange.Start.Line == sourceMappingRange.End.Line)
```

Comment:
It turns out just writing "If on type formatting is happening on a single line" in a comment, doesn't actually mean the code will only run in that scenario.

Who knew!

🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️

Reply:
I dono if sourceMappings suffer from this but in my current formatting work you have to worry about whether the node you're working on includes the trailing whitespace (which may be a newline). Just something to consider/test.

Reply:
Definitely relevant here. The `newLineCount` comes from looking at the whitespace after the transition, and if there are any newlines in it.